ha_lineairdb.cc

```diff
@@ -870,82 +844,107 @@ bool ha_lineairdb::store_blob_to_field(Field** field) {
-   */
-  my_bitmap_map* org_bitmap = dbug_tmp_use_all_columns(table, table->write_set);
-
   // Clear BLOB data from the previous row.
   blobroot.ClearForReuse();
-
-  std::byte* p                  = (std::byte*)malloc(read_buf_size);
-  const std::byte* const init_p = p;
-
-  memcpy(p, read_buf, read_buf_size);
-  std::byte* buf_end = p + read_buf_size;
-
-  /**
-   * extract values from LineairDB read_buf and 
-   * store each column value to corresponding field
-  */
-  for (Field** field = table->field; *field; field++) {
-    write_buffer_.length(0);
-    for (; p < buf_end; p++) {
-      uchar c              = *reinterpret_cast<uchar*>(p);
-      bool is_end_of_field = p == buf_end - 1 ? true : false;
-      switch (c) {
-        case '\"':
-          break;
-        case ',':
-          is_end_of_field = 1;
-          break;
-        default:
-          write_buffer_.append(c);
-          break;
-      }
-      if (is_end_of_field || p == buf_end) {
-        (*field)->store(write_buffer_.ptr(), write_buffer_.length(),
-                          write_buffer_.charset(), CHECK_FIELD_WARN);
-        if (store_blob_to_field(field)) return HA_ERR_OUT_OF_MEM;
-        p++;
-        break;
-      }
-    }
-  }
-
-  write_buffer_.length(0);
+  ldbField.make_mysql_table_row(read_buf, read_buf_size);
   /**
    * for each 8 potentially null columns, buf holds 1 byte flag at the front
-   * the number of null flag bytes in buf is shown in table->s->nullbytes
+   * the number of null flag bytes in buf is shown in table->s->null_bytes
    * the flag is originally set to 0xff, or b11111111
    * if you want to make the first potentially null column to show a non-null
    * value, store 0xfe, or b11111110, in buf
   */
-  translator.set_null_flags_in_buf(buf);
+  auto nullFlags = ldbField.get_null_flags();
+  for (size_t i = 0; i < nullFlags.size(); i++) { buf[i] = nullFlags[i]; }
 
-  free((void*)init_p);
+  /* Avoid asserts in ::store() for columns that are not going to be updated
+   */
+  my_bitmap_map* org_bitmap = dbug_tmp_use_all_columns(table, table->write_set);
+  /**
+   * store each column value to corresponding field
+  */
+  size_t columnIndex = 0;
+  for (Field** field = table->field; *field; field++) {
+    const auto mysqlFieldValue = ldbField.get_column_of_row(columnIndex++);
+    (*field)->store(mysqlFieldValue.c_str(), mysqlFieldValue.length(),
+                    &my_charset_bin, CHECK_FIELD_WARN);
+    if (store_blob_to_field(field)) return HA_ERR_OUT_OF_MEM;
+  }
   dbug_tmp_restore_column_map(table->write_set, org_bitmap);
   return 0;
 }
+
+/**
+ * Lineairdb_Field method definitions
+*/
+
+std::string Lineairdb_Field::convert_numeric_to_bytes(const size_t num) {
+  char bytes[sizeof(size_t)];
+  std::copy(reinterpret_cast<const char*>(&num),
+            reinterpret_cast<const char*>(&num) + sizeof(size_t),
+            bytes);
+  return std::string{bytes};
```

Comment:
Could you write it as one line? or remove `bytes`?
I think there is unnecessary copying from the local variable (char) to the string.
The cost of copying string is expensive and unbounded.